tests/bugs/gh-3087-multi-entry-point.slang

```diff
@@ -5,12 +5,12 @@
 
 // we should only have 1 'BuiltIn InstanceIndex' since the `Output` and `Input` semantic
 // for `InstanceIndex` should be converted to a non-builtin
-// CHECK-DAG: BuiltIn InstanceIndex
-// CHECK-NOT: BuiltIn InstanceIndex
+// CHECK: OpDecorate %gl_InstanceIndex BuiltIn InstanceIndex
+// CHECK-NOT: OpDecorate %gl_InstanceIndex BuiltIn InstanceIndex
```

Comment:
```suggestion
// CHECK-DAG: OpDecorate %gl_InstanceIndex BuiltIn InstanceIndex
// CHECK-NOT: OpDecorate %gl_InstanceIndex BuiltIn InstanceIndex
```
It is possible we emit the `Flat` decoration before the `OpDecorate %gl_InstanceIndex BuiltIn InstanceIndex` decoration, I think it may be safer to add a `CHECK-DAG` check here.